hasher-matcher-actioner/hmalib/fetching/bank_store.py

```diff
@@ -0,0 +1,289 @@
+                self._unflushed_updates[update_key] = update_value
+                self._unflushed_deletes.discard(update_key)
+
+    def flush(self, checkpoint: FetchCheckpointBase) -> None:
+        if self._dirty:
+            # self._delta check is avoidable as it is covered by self._dirty,
+            # but python type checker gets annoyed.
+
+            del_count = 0
+            while len(self._unflushed_deletes) > 0:
+                delete = self._unflushed_deletes.pop()
+                self._handle_delete(delete)
+                del_count = del_count + 1
+            logger.info(
+                "BankCollabFetchStore flushed %d deletes for collab: %s",
+                del_count,
+                self.collab.name,
+            )
+
+            upsert_count = 0
+            for update_key in list(self._unflushed_updates.keys()):
+                update_value = self._unflushed_updates.pop(update_key)
+                self._handle_upsert(update_key, update_value)
+                upsert_count = upsert_count + 1
+            logger.info(
+                "BankCollabFetchStore flushed %d upserts for collab: %s",
+                upsert_count,
+                self.collab.name,
+            )
+        else:
+            logger.info("FetchStore is clean!")
+
+        logger.info("Setting checkpoint for collab: %s", self.collab.name)
+        self.set_checkpoint(checkpoint)
+
+    def clear(self) -> None:
+        """
+        This one might be a time-consuming operation.
+
+        The ideal way would be to get a list of all bank members and then remove
+        them and their signals. However, that would have to rely on paginated
+        queries and batch operations which are both slow in dynamodb.
+
+        An easier way out would be to iterate over all collab configs, unlink
+        their banks and then
+        """
+        raise NotImplementedError
+
+    def _handle_upsert(self, key: TUpdateRecordKey, value: TUpdateRecordValue):
+        """
+        Respond to an update event from a signal exchange API. This can happen
+        at most once per key per flush.
+
+        A single update can translate into multiple signals, but all will be
+        stored as part of the same bank-member.
+        """
+        str_key = self._coerce_key(key)
+
+        # Call naive_convert_to_signal_type one update at a time, to get the
+        # full expected state of a bank-member.
+        signal_type_view = self._api_cls.naive_convert_to_signal_type(
+            self._signal_types, self.collab.to_pytx_collab_config(), {key: value}
+        )
+
+        if len(signal_type_view) == 0:
+            # Cases where none of the signal types in the update can be
+            # processed. eg. NCMEC returning pdna
+            return
+
+        # Ascertain that all signal_types map from the same content type.
+        # HMA at present can't store multiple content types in one bank
+        # member.
```

Comment:
blocking: Seems like a very important part of the implementation! Users generally have the expectation that as long as they follow the interface rules, that implementers should be able to handle it. 

if not, we should tighten down either the interface or the implementation. In this case, I think there might be a hacky solution that would handle. It could also be that we just need to add an "other" case that simply doesn't display on the UI, but still allows HMA to process it.

Reply:
boop, did I miss it in your new docstring?

Reply:
TODO: #1221